lib/pacemaker/pcmk_rule.c

```diff
@@ -126,15 +126,15 @@ rule_init(pcmk__output_t *out, xmlNodePtr input, crm_time_t *date,
  * \return Standard Pacemaker return code
  */
 static int
-crm_rule_check(pcmk__output_t *out, pe_working_set_t *data_set,
-               const char *rule_id, crm_time_t *date)
+do_check_rule(pcmk__output_t *out, pe_working_set_t *data_set,
```

Comment:
I know @kgaillot isn't really a fan of `do_` functions.

Reply:
Fair enough I thought `crm_rule_check` looked too much like a public name, and a bare `rule_check` or `check_rule` could be confusing when we already have `pcmk__rule_check` and `pcmk_rule_check` (the inline wrappers).

Reply:
Maybe `check_rule_effectiveness` or `eval_rule`

Reply:
`eval_rule` seems like a good fit since `eval` is the pattern for this kind of thing in `rules.c` (e.g., `pe_eval_rules`, `pe_eval_expr`). It's almost too generic since we might want to use `eval_rule` in something like `rules.c` later -- but it's static so we can change it if we ever need to. (Static name conflicts won't break compilation, just good to avoid for logging purposes).